x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -263,6 +268,19 @@ export const InnerWorkspacePanel = React.memo(function InnerWorkspacePanel({
               searchService: plugins.data.search,
             }
           );
+          esTookTime.current = adapters.requests.getRequests().reduce((maxTime, { response }) => {
+            if (response === undefined) {
+              return maxTime;
+            }
+            const took =
+              (
+                response.json as {
+                  rawResponse: estypes.SearchResponse | undefined;
+                }
+              )?.rawResponse?.took ?? 0;
+
+            return Math.max(maxTime, took);
```

Comment:
I think with an additional optional chaining it's possible to compact even further:
```suggestion
            const took =
              (
                response?.json as {
                  rawResponse: estypes.SearchResponse | undefined;
                }
              )?.rawResponse?.took ?? 0;

            return Math.max(maxTime, took);
```

Reply:
Nice catch, but it doesn't change much, because looking at what I wrote, I need to fix it l like this:

```suggestion
            const took =
              (
                response?.json as
                  | {
                      rawResponse: estypes.SearchResponse | undefined;
                    }
                  | undefined
              )?.rawResponse?.took ?? 0;

            return Math.max(maxTime, took);
```

The `as` typescript casting casts the `response?.json` to something that exists as `{rawResponse: estypes.SearchResponse | undefined }`, but in reality this can be also `undefined`. 
The optional chain after the cast (that I wrongly added) is actually optional with that signature, creating an issue if someone gets rid of unnecessary optional chaining automatically (lining or just by hand).

In both case I don't see any special benefit of going in one or the other way, in each case is not a great read.
Initially I tried to get rid of that completely (see [885211a](https://github.com/elastic/kibana/pull/192245/commits/885211a768b03f34ec17a1a96306048d315bacff)) by using a correct type for the `json` parameter, but there was a circular dependency that I didn't tried to fix as it was a bit too much for that

Reply:
done here https://github.com/elastic/kibana/pull/192245/commits/0cce1fab80d11fce39167e1f95a177ebbcbafa5a